cmakebuild.js

```diff
@@ -0,0 +1,48 @@
+var util = require('./util')
+var spawn = require('child_process').spawn
+var which = require('npm-which')(process.cwd())
+
+function runCmake (opts, target, cb) {
+  var log = opts.log
+
+  which('cmake-js', function (err, cmakeJsPath) {
+    if (err) return cb(err)
+
+    if (!opts.preinstall) return run(cmakeJsPath)
+
+    log.verbose('executing preinstall')
+    util.exec(opts.preinstall, function (err) {
```

Comment:
This preinstall code is in two places and should be moved out to `build.js`, which means `gypbuild.js` can be simpler (can remove the internal `run` function and reduce scope.